python/ray/_private/job_manager/job_manager.py

```diff
@@ -207,64 +246,88 @@ def _get_current_node_resource_key(self) -> str:
-            # running on headnode, same for job supervisor actors scheduled
-            resources={
-                self._get_current_node_resource_key(): 0.001,
-            },
-            # For now we assume supervisor actor and driver script have same
-            # runtime_env.
-            runtime_env=runtime_env,
-        ).remote(job_id, metadata or {})
+        self._status_client.put_status(job_id, JobStatus.PENDING)
+        supervisor = None
 
         try:
+            logger.debug(
+                f"Submitting job with generated internal job_id: {job_id}")
+            supervisor = self._supervisor_actor_cls.options(
+                lifetime="detached",
+                name=self.JOB_ACTOR_NAME.format(job_id=job_id),
+                # Currently we assume JobManager is created by dashboard server
+                # running on headnode, same for job supervisor actors scheduled
+                resources={
+                    self._get_current_node_resource_key(): 0.001,
+                },
+                # For now we assume supervisor actor and driver script have
+                # same runtime_env.
+                runtime_env=runtime_env).remote(job_id, metadata or {})
             ray.get(
                 supervisor.ready.remote(), timeout=self.START_ACTOR_TIMEOUT_S)
-        except GetTimeoutError:
-            ray.kill(supervisor, no_restart=True)
-            raise RuntimeError(f"Failed to start actor for job {job_id}.")
+        except Exception as e:
+            if supervisor:
+                ray.kill(supervisor, no_restart=True)
+            self._status_client.put_status(job_id, JobStatus.FAILED)
+            raise RuntimeError(
+                f"Failed to start actor for job {job_id}. This could be "
+                "runtime_env configuration failure, or timed out after "
+                f"{self.START_ACTOR_TIMEOUT_S} secs. "
+                f"Exception message: {str(e)}")
 
         # Kick off the job to run in the background.
-        supervisor.run.remote(entrypoint)
+        supervisor.run.remote(entrypoint, _start_signal_actor)
 
         return job_id
 
     def stop_job(self, job_id) -> bool:
-        """Request job to exit."""
+        """Request job to exit.
+
+        Returns:
+            stopped (bool):
+                True if there's running actor job we intend to stop
+                False if no running actor for the job found
+        """
         job_supervisor_actor = self._get_actor_for_job(job_id)
         if job_supervisor_actor is not None:
-            # Actor is still alive, signal it to stop the driver.
+            # Actor is still alive, signal it to stop the driver, fire and
+            # forget
             job_supervisor_actor.stop.remote()
+            return True
+        else:
+            return False
 
     def get_job_status(self, job_id: str):
         job_supervisor_actor = self._get_actor_for_job(job_id)
```

Comment:
We should only be doing this check if the job is in a non-terminal state (`PENDING`, `RUNNING`)

Reply:
Otherwise we know the actor is gone, no need for this